ansible/library/announce_routes.py

```diff
@@ -87,16 +87,18 @@ def announce_routes(ptf_ip, port, routes):
 def generate_routes(family, podset_number, tor_number, tor_subnet_number,
                     spine_asn, leaf_asn_start, tor_asn_start,
                     nexthop, nexthop_v6,
-                    tor_subnet_size, max_tor_subnet_number,
-                    router_type = "leaf", tor_index=None):
+                    tor_subnet_size, max_tor_subnet_number, topo,
+                    router_type = "leaf", tor_index=None, set_num=None):
     routes = []
 
     default_route_as_path = "6666 6667"
 
-    if router_type == "leaf":
+    if router_type == "leaf" and topo != "t2":
         default_route_as_path = "{} {}".format(spine_asn, default_route_as_path)
-
-    if router_type != 'tor':
+    if router_type == "core" and topo == "t2":
+        default_route_as_path = "{}".format(default_route_as_path)
+    # Default route for T2 is adventised by the core routers.
+    if (topo != "t2" and router_type != 'tor') or (topo == "t2" and router_type == "core"):
```

Comment:
This whole if condition means "up link router", right? can you make it a helper function? It would be easier to maintain in the future with a properly named helper function and spell out all 3 conditions: leaf of T0, spine of T1 and core of T2.